### DISCUSSION TITLE
Trouble running the latest version (`1.32.2`) on FreeBSD (TrueNAS)

### DISCUSSION THREAD

**Discussion by danbee** [dani-garcia/vaultwarden#5119] [Q&A]
I was running `1.32.0` on my TrueNAS server (installed with `pkg`), but then I started running into issues with the iOS app which led me to attempt an upgrade. Unfortunately `1.32.0` is the latest version currently available in `pkg`, so I decided to install the port instead, as that is at `1.32.2`.

It works just fine if I run it from the shell, and have setup a `bash` script to start the server from within the `/usr/local/www/vaultwarden` directory.

I cannot get it to start with the init script though. And I can't seem to get any output that's helpful in debugging the issue. If I try and start it with `rc_debug=yes` I get the following output. It looks like something is going on with the PID but I'm not sure what it is. Anyway, if I check the process list after running this I can see that the `vaultwarden` process is not running.

```
# service vaultwarden start
/usr/local/etc/rc.d/vaultwarden: DEBUG: Sourcing /etc/defaults/rc.conf
/usr/local/etc/rc.d/vaultwarden: DEBUG: Sourcing /usr/local/etc/rc.conf.d/vaultwarden
/usr/local/etc/rc.d/vaultwarden: DEBUG: pid file (/var/run/vaultwarden/vaultwarden.pid): no pid in file.
/usr/local/etc/rc.d/vaultwarden: DEBUG: checkyesno: vaultwarden_enable is set to YES.
/usr/local/etc/rc.d/vaultwarden: DEBUG: run_rc_command: start_precmd: start_precmd
Starting vaultwarden.
/usr/local/etc/rc.d/vaultwarden: DEBUG: run_rc_command: doit: cd /usr/local/www/vaultwarden &&  limits -C daemon  su -m www -c 'sh -c "/usr/sbin/dae
mon  -p /var/run/vaultwarden/vaultwarden.pid -T vaultwarden /usr/local/bin/vaultwarden"'
```

**Comment by BlackDex** [comment]
I'm not seeing logs from Vaultwarden it self. I would expect to see some backtrace from Vaultwarden or an executable/elf error.

I'm not seeing both. 
Try to start Vaultwarden manually instead of the services handler and see if you get more info.

**Reply by danbee** [reply]
Thanks for your reply @BlackDex, but if you re-read my original post you'll see that I've already tried running it manually and it runs just fine. I've tried running it via `daemon` like the init script does and it just exits without any output and a successful error code.

```
# daemon -p /var/run/vaultwarden/vaultwarden.pid -T vaultwarden /usr/local/bin/vaultwarden
# echo $?
0
```

**Comment by danbee** [comment]
More info: I mentioned in my original post that I've tried running the server manually and it works just fine. So it's either a problem with the init script or a problem running it via `daemon`.

```
# ./start
/--------------------------------------------------------------------\
|                        Starting Vaultwarden                        |
|--------------------------------------------------------------------|
| This is an *unofficial* Bitwarden implementation, DO NOT use the   |
| official channels to report bugs/features, regardless of client.   |
| Send usage/configuration questions or feature requests to:         |
|   https://github.com/dani-garcia/vaultwarden/discussions or        |
|   https://vaultwarden.discourse.group/                             |
| Report suspected bugs/issues in the software itself at:            |
|   https://github.com/dani-garcia/vaultwarden/issues/new            |
\--------------------------------------------------------------------/

[2024-10-22 11:44:53.570][start][INFO] Rocket has launched from http://192.168.42.10:4567
```

**Reply by danbee** [reply]
The `start` script looks like this:

```bash
#!/usr/local/bin/bash

IP_HEADER=X-Forwarded-For \
ROCKET_ADDRESS=192.168.42.10 \
ROCKET_PORT=4567 \
SIGNUPS_ALLOWED='false' \
DOMAIN='https://{redacted}' \
ADMIN_TOKEN='{redacted}' \
SMTP_HOST=localhost \
SMTP_FROM=noreply@localhost \
SMTP_PORT=587 \
SMTP_SECURITY=starttls \
vaultwarden
```

**Reply by BlackDex** [reply]
Then i would suggest to contact the maintainers of the FreeBSD package of Vaultwarden and see if they have any idea on how to troubleshoot this as I'm no *BSD expert.

**Comment by sporkman** [comment]
I have no issues running this on FreeBSD (13.3), just upgraded to the same version. TrueNAS however, is a slightly different beast, so you might want to look in their direction for help.

**Reply by danbee** [reply]
It's running in a jail so I just assumed it was a standard FreeBSD base. It's running on 13.1, although `iocage` believes it to be 13.4. Thanks for the datapoint though. I'll ask the TrueNAS folks.

**Reply by danbee** [reply]
Can I ask what your `init` script looks like? Mine looks like this:

`/usr/local/etc/rc.d/vaultwarden`
```sh
#!/bin/sh

# PROVIDE: vaultwarden
# REQUIRE: LOGIN DAEMON NETWORKING
# KEYWORD: rust

# Enable this script by adding:
# vaultwarden_enable="YES"
# ... to /etc/rc.conf
rc_debug=yes

. /etc/rc.subr
name=vaultwarden

rcvar=vaultwarden_enable
load_rc_config ${name}

: ${vaultwarden_chdir=/usr/local/www/vaultwarden}
: ${vaultwarden_user:=www}
: ${vaultwarden_group:=www}

# This is the tool init launches
command="/usr/sbin/daemon"

pidfile="/var/run/${name}/${name}.pid"

task="/usr/local/bin/${name}"
procname="${task}"

command_args="-p ${pidfile} -T ${name} ${task}"

start_precmd="start_precmd"
start_precmd()
{
        if [ ! -e "/var/run/${name}" ] ; then
                install -d -o ${vaultwarden_user} -g ${vaultwarden_group} /var/run/${name};
        fi

        # checkyesno rubywarden_api_signups
        # if [ "$?" -eq 0 ]; then
        #         export ALLOW_SIGNUPS=1
        #         echo "Vaultwarden Signups Enabled"
        # fi
}

run_rc_command "$1"
```
